### PR TITLE
refactor: move source code into a src folder

### PR DESCRIPTION
**Pull request by chaojun-zhang**
move source code into a src folder

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,4 +30,4 @@ if(${SUBSTRAIT_CPP_BUILD_TESTING})
   enable_testing()
 endif()
 
-add_subdirectory(substrait)
+add_subdirectory(src/substrait)
```
Comment (jbalint) [inline CMakeLists.txt:33]: is there any value in keeping subtrait/ under src/? why not just rename substrait to src?

Comment (EpsilonPrime) [reply CMakeLists.txt:33]: I'm currently importing substrait internal headers with:

`#include "substrait/textplan/parser/ParseResult.h"`

If we only do a rename the fact that a substrait file is being included goes away.  Also helps to mirror the path hierarchy (although we have an extra ::io in this module).

Comment (chaojun-zhang) [reply CMakeLists.txt:33]: Agreed, the key is we can refer header files in a  consistent way that always start with `substrait` prefix, whether it is either under src/substrait or under include/substrait directory.

Comment (jbalint) [reply CMakeLists.txt:33]: Makes sense. Thanks for the explanation

**Review by EpsilonPrime — APPROVED** [review]
Looks good to me!

**Review by westonpace — APPROVED** [review]
Thanks!  LGTM

**Pull request merged**